### PR TITLE
[BEAM-2213] Convert Executor Services to use Daemon Threads

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`.
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
This will cause the DirectRunner to automatically shut down when the
worker threads are shut down.

This is mostly a fork of https://github.com/apache/beam/pull/2936, with a couple of additional executors changed and non-ExecutorService work removed.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @dhalperi

File: runners/direct-java/src/main/java/org/apache/beam/runners/direct/ExecutorServiceParallelExecutor.java [hunk 1]
```diff
@@ -132,7 +134,13 @@ private ExecutorServiceParallelExecutor(
...
       EvaluationContext context) {
     this.targetParallelism = targetParallelism;
-    this.executorService = Executors.newFixedThreadPool(targetParallelism);
+    this.executorService =
+        Executors.newFixedThreadPool(
+            targetParallelism,
+            new ThreadFactoryBuilder()
+                .setThreadFactory(MoreExecutors.platformThreadFactory())
+                .setNameFormat("direct-runner-worker")
+                .build());
```
Comment (dhalperi) [inline runners/direct-java/src/main/java/org/apache/beam/runners/direct/ExecutorServiceParallelExecutor.java]: add a comment why this one is not a daemon?

Comment (tgroh) [inline runners/direct-java/src/main/java/org/apache/beam/runners/direct/ExecutorServiceParallelExecutor.java]: Done.

**Comment by dhalperi** [comment]
[ERROR] /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_MavenInstall/runners/direct-java/src/main/java/org/apache/beam/runners/direct/SplittableProcessElementsEvaluatorFactory.java:176: Line is longer than 100 characters (found 106). [LineLength]

**Pull request closed without merging**